tf_agents/utils/nest_utils.py

```diff
@@ -554,5 +560,9 @@ def where(condition, true_outputs, false_outputs):
     Interleaved output from `true_outputs` and `false_outputs` based on
     `condition`.
   """
-  return tf.nest.map_structure(lambda t, f: tf.compat.v1.where(condition, t, f),
-                               true_outputs, false_outputs)
+  return tf.nest.map_structure(
+    lambda t, f: tf.where(
+      tf.reshape(condition,
+                 tf.concat([tf.shape(condition), tf.ones(tf.rank(t) - tf.rank(condition), dtype=tf.int32)], axis=0)),
```

Comment:
i wonder if there's a cleaner way to rewrite this part here so its not all in one line? maybe define a helper _new_shape() and pass that to reshape?

Reply:
Hi @egonina . I've converted the lambda into a function and broken it into multiple lines for clarity per your suggestion.